src/libostree/ostree-repo-checkout.c

```diff
@@ -1055,7 +1053,7 @@ checkout_tree_at_recurse (OstreeRepo *self, OstreeRepoCheckoutAtOptions *options
     if (sepolicy_enabled && _ostree_sepolicy_host_enabled (options->sepolicy))
       {
         /* We'll set the xattr via setfscreatecon(), so don't do it via generic xattrs below. */
-        modified_xattrs = _ostree_filter_selinux_xattr (xattrs);
+        GVariant *modified_xattrs = _ostree_filter_selinux_xattr (xattrs);
         xattrs = modified_xattrs;
```

Comment:
*This* patch is superficially similar to the above but very different in that we very clearly have two `g_autoptr` pointing to the same thing which is obviously wrong.

However - we're now leaking memory for the old value of `xattrs`. What we can do is instead:

```
g_autoptr(GVariant) old_xattrs = g_steal_pointer (&xattrs);
xattrs = _ostree_filter_selinux_xattr (old_xattrs);
```
or so.

(And yes, this is all incredibly subtle and easy to get wrong and C sucks...)